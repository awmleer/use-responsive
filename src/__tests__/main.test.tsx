import * as testing from '@testing-library/react'
import {FC, useState} from 'react'
import * as React from 'react'
import {useResponsive} from '../index'
import {act} from 'react-dom/test-utils'

test('response to window width changes', function () {
  act(() => {
    //@ts-ignore
    global.innerWidth = 500
    //@ts-ignore
    global.dispatchEvent(new Event('resize'))
  })
  function App() {
    const width = useResponsive()
    return (
      <p>{width}</p>
    )
  }
  const renderer = testing.render(
    <App/>
  )
  expect(renderer.asFragment()).toMatchSnapshot()
  act(() => {
    //@ts-ignore
    global.innerWidth = 1000
    //@ts-ignore
    global.dispatchEvent(new Event('resize'))
  })
  expect(renderer.asFragment()).toMatchSnapshot()
})
