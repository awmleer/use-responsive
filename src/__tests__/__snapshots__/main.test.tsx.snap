// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`response to window width changes 1`] = `
<DocumentFragment>
  <p>
    500
  </p>
</DocumentFragment>
`;

exports[`response to window width changes 2`] = `
<DocumentFragment>
  <p>
    1000
  </p>
</DocumentFragment>
`;
